test/common/scale_up_replicas_threescale.go

```diff
@@ -0,0 +1,166 @@
+	if err := checkNumberOfReplicasAgainstValue(apim, ctx, numberOfReplicas, retryInterval, timeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+}
+
+func getAPIManager(dynClient k8sclient.Client) (threescalev1.APIManager, error) {
+	apim := &threescalev1.APIManager{}
+
+	if err := dynClient.Get(goctx.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, apim); err != nil {
+		return *apim, err
+	}
+
+	return *apim, nil
+}
+
+func updateAPIManager(dynClient k8sclient.Client, replicas int64) (threescalev1.APIManager, error) {
+
+	apim, err := getAPIManager(dynClient)
+	if err != nil {
+		return apim, err
+	}
+	resourceRequirements := true
+	apim = threescalev1.APIManager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name,
+			Namespace:       namespace,
+			ResourceVersion: apim.GetResourceVersion(),
+		},
+		Spec: threescalev1.APIManagerSpec{
+			HighAvailability: &threescalev1.HighAvailabilitySpec{},
+			APIManagerCommonSpec: threescalev1.APIManagerCommonSpec{
+				ResourceRequirementsEnabled: &resourceRequirements,
+			},
+			System: &threescalev1.SystemSpec{
+				DatabaseSpec: &threescalev1.SystemDatabaseSpec{
+					PostgreSQL: &threescalev1.SystemPostgreSQLSpec{},
+				},
+				FileStorageSpec: &threescalev1.SystemFileStorageSpec{
+					S3: &threescalev1.SystemS3Spec{},
+				},
+				AppSpec:     &threescalev1.SystemAppSpec{Replicas: &[]int64{replicas}[0]},
+				SidekiqSpec: &threescalev1.SystemSidekiqSpec{Replicas: &[]int64{replicas}[0]},
+			},
+			Apicast: &threescalev1.ApicastSpec{
+				ProductionSpec: &threescalev1.ApicastProductionSpec{Replicas: &[]int64{replicas}[0]},
+				StagingSpec:    &threescalev1.ApicastStagingSpec{Replicas: &[]int64{replicas}[0]},
+			},
+			Backend: &threescalev1.BackendSpec{
+				ListenerSpec: &threescalev1.BackendListenerSpec{Replicas: &[]int64{replicas}[0]},
+				WorkerSpec:   &threescalev1.BackendWorkerSpec{Replicas: &[]int64{replicas}[0]},
+				CronSpec:     &threescalev1.BackendCronSpec{Replicas: &[]int64{replicas}[0]},
+			},
+			Zync: &threescalev1.ZyncSpec{
+				AppSpec: &threescalev1.ZyncAppSpec{Replicas: &[]int64{replicas}[0]},
+				QueSpec: &threescalev1.ZyncQueSpec{Replicas: &[]int64{replicas}[0]},
+			},
+		},
+	}
+
+	if err := dynClient.Update(goctx.TODO(), apim.DeepCopy(), &k8sclient.UpdateOptions{}); err != nil {
+		return apim, err
+	}
+
+	return apim, nil
+}
+
+func checkNumberOfReplicasAgainstValue(apim threescalev1.APIManager, ctx *TestingContext, numberOfRequiredReplicas int64, retryInterval, timeout time.Duration, t *testing.T) error {
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		apim, err = getAPIManager(ctx.Client)
+		if err != nil {
+			t.Fatalf("failed to get APIManager : %v", err)
+		}
+		if *apim.Spec.System.AppSpec.Replicas != numberOfRequiredReplicas {
+			return false, fmt.Errorf("Number of replicas for apim.Spec.System.AppSpec is not correct : %v", *apim.Spec.System.AppSpec.Replicas)
```

Comment:
```suggestion
			return false, fmt.Errorf("Number of replicas for apim.Spec.System.AppSpec is not correct : Replicas - %v, Expected - %v", *apim.Spec.System.AppSpec.Replicas, numberOfRequiredReplicas)
```
Would be good to also list the expected replicas in this output.